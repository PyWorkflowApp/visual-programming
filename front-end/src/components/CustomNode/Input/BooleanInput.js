import React, { useState, useEffect } from 'react';
import { Form } from 'react-bootstrap';

export default function BooleanInput(props) {

    const [value, setValue] = useState(props.value);
    const handleChange = (event) => {
        setValue(event.target.checked);
    };

    const {keyName, onChange} = props;
    // whenever value changes, fire callback to update config form
    useEffect(() => {
            onChange(keyName, value);
        },
        [value, keyName, onChange]);

    return  (
        <Form.Check type="checkbox" name={props.keyName}
                    disabled={props.disabled}
                    checked={value}
                    onChange={handleChange} />
    )
}
