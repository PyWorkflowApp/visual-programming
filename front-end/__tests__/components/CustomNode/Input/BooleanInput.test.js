import React from 'react'
import { shallow, mount } from 'enzyme';
import BooleanInput from '../../../../src/components/CustomNode/Input/BooleanInput';

describe('Validates BooleanInpupt', () => {
  it('Display BooleanInput', () => {
    const booleanInput = shallow(<BooleanInput />);
    expect(booleanInput).toMatchSnapshot();
  });

  it('Checked BooleanInput', () => {

    const onChange = jest.fn(() => []);
    const booleanInput = shallow(<BooleanInput
                        keyName="booleanSelector"
                        value={false}
                        onChange={onChange}/>);

    expect(booleanInput.find({name: "booleanSelector"}).prop('checked')).toBe(false);

    booleanInput.find({name: "booleanSelector"}).simulate('change', {target: {checked: true}})

    booleanInput.update();
    expect(booleanInput.find({name: "booleanSelector"}).prop('checked')).toBe(true);
  });

  it('Checked BooleanInput onChange', () => {

    const onChange = jest.fn(() => []);
    const booleanInput = mount(<BooleanInput
                        keyName="booleanSelector"
                        value={false}
                        onChange={onChange}/>);

    booleanInput.find({keyName: "booleanSelector"}).simulate('change', {target: {checked: true}})

    expect(onChange.mock.calls.length).toBe(1);
  });
});
