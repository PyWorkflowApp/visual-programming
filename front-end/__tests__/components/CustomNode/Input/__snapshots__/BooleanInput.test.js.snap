// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validates BooleanInpupt Display BooleanInput 1`] = `ShallowWrapper {}`;
